v2/yarpcthrift/thriftrw-plugin-yarpc2/main.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// thriftrw-plugin-yarpc implements a plugin for ThriftRW that generates code
+// compatible with YARPC.
+//
+// For more information, check the documentation of the parent package.
+package main
+
+import (
+	"flag"
+	"strings"
+
+	"go.uber.org/thriftrw/plugin"
+	"go.uber.org/thriftrw/plugin/api"
+)
+
+// Command line flags
+var (
+	_context = flag.String("context-import-path",
+		"context",
+		"Import path at which Context is available")
+	_unaryHandlerWrapper = flag.String("unary-handler-wrapper",
+		"go.uber.org/yarpc/v2/yarpcthrift.Handler",
+		"Function used to wrap generic Thrift unary function handlers into YARPC handlers")
+	_noGomock = flag.Bool("no-gomock", false,
+		"Don't generate gomock mocks for service clients")
+	_noFx             = flag.Bool("no-fx", false, "Don't generate Fx module")
+	_sanitizeTChannel = flag.Bool("sanitize-tchannel", false, "Enable tchannel context sanitization")
+)
+
+type g struct {
+	SanitizeTChannel bool
+}
+
+func (g g) Generate(req *api.GenerateServiceRequest) (*api.GenerateServiceResponse, error) {
+	generators := []genFunc{clientGenerator, serverGenerator}
+	if !*_noFx {
```

Comment:
Given the stance we've taken for the `protoc-gen-yarpc-go` plugin in v2, we might not need this flag in the long run. No need to remove it here, but something worth noting.

Reply:
Sorry, what was the stance we took in `protoc-gen-yarpc-go`? Are we always going to generate fx-related files or is it changing to be an opt-in?

Reply:
The stance was that we could assume Fx is available, which leans towards always having an fx module generated. However, this is fine for now. We'll probably want to look over all of the flags declared here in a follow up.